Author: Chris Nolan <chris@cenolan.com>
Description:
 Allows the driver to associate with hidden SSID networks using Network Manager.
 http://worldofxor.blogspot.com/2008/12/it-works-broadcom-official-wireless.html

diff -Naurp broadcom-sta.orig/amd64/src/wl/sys/wl_iw.c broadcom-sta/amd64/src/wl/sys/wl_iw.c
--- broadcom-sta.orig/amd64/src/wl/sys/wl_iw.c	2009-05-03 09:16:11.000000000 +0000
+++ broadcom-sta/amd64/src/wl/sys/wl_iw.c	2009-05-17 14:23:45.000000000 +0000
@@ -580,6 +580,7 @@ wl_iw_get_range(
 	range->enc_capa |= IW_ENC_CAPA_CIPHER_TKIP;
 	range->enc_capa |= IW_ENC_CAPA_CIPHER_CCMP;
 	range->enc_capa |= IW_ENC_CAPA_WPA2;
+	range->scan_capa = IW_SCAN_CAPA_ESSID;
 #endif 
 
 	return 0;
diff -Naurp broadcom-sta.orig/i386/src/wl/sys/wl_iw.c broadcom-sta/i386/src/wl/sys/wl_iw.c
--- broadcom-sta.orig/i386/src/wl/sys/wl_iw.c	2009-05-03 09:16:11.000000000 +0000
+++ broadcom-sta/i386/src/wl/sys/wl_iw.c	2009-05-17 14:23:48.000000000 +0000
@@ -580,6 +580,7 @@ wl_iw_get_range(
 	range->enc_capa |= IW_ENC_CAPA_CIPHER_TKIP;
 	range->enc_capa |= IW_ENC_CAPA_CIPHER_CCMP;
 	range->enc_capa |= IW_ENC_CAPA_WPA2;
+	range->scan_capa = IW_SCAN_CAPA_ESSID;
 #endif 
 
 	return 0;
